src/TLuaInterpreter.cpp

```diff
@@ -6081,111 +6062,98 @@ int TLuaInterpreter::exists(lua_State* L)
                 cnt++;
             }
             it1++;
         }
-    } else if (type == "trigger") {
+    } else if (type.compare(QLatin1String("trigger"), Qt::CaseInsensitive) == 0) {
         QMap<QString, TTrigger*>::const_iterator it1 = host.getTriggerUnit()->mLookupTable.constFind(name);
         while (it1 != host.getTriggerUnit()->mLookupTable.cend() && it1.key() == name) {
             if (it1.value()->isActive()) {
                 cnt++;
             }
             it1++;
         }
-    } else if (type == "alias") {
+    } else if (type.compare(QLatin1String("alias"), Qt::CaseInsensitive) == 0) {
         QMap<QString, TAlias*>::const_iterator it1 = host.getAliasUnit()->mLookupTable.constFind(name);
         while (it1 != host.getAliasUnit()->mLookupTable.cend() && it1.key() == name) {
             if (it1.value()->isActive()) {
                 cnt++;
             }
             it1++;
         }
-    } else if (type == "keybind") {
+    } else if (type.compare(QLatin1String("keybind"), Qt::CaseInsensitive) == 0) {
         QMap<QString, TKey*>::const_iterator it1 = host.getKeyUnit()->mLookupTable.constFind(name);
         while (it1 != host.getKeyUnit()->mLookupTable.cend() && it1.key() == name) {
             if (it1.value()->isActive()) {
                 cnt++;
             }
             it1++;
         }
+    } else {
+        lua_pushnil(L);
+        lua_pushfstring(L, "invalid type '%s' given, it should be one (case insensitive) of: 'alias', 'keybind', 'timer' or 'trigger'",
+                        type.toUtf8().constData());
     }
     lua_pushnumber(L, cnt);
     return 1;
 }
 
-
 int TLuaInterpreter::permAlias(lua_State* L)
 {
-    string luaName;
     if (!lua_isstring(L, 1)) {
-        lua_pushstring(L, "permAlias: need a name for this alias");
-        lua_error(L);
-        return 1;
-    } else {
-        luaName = lua_tostring(L, 1);
+        lua_pushfstring(L, "permAlias: bad argument #1 type (alias name as string expected, got %s!)",
+                        luaL_typename(L, 1));
+        return lua_error(L);
     }
+    QString name = QString::fromUtf8(lua_tostring(L, 1));
 
-    string luaParent;
     if (!lua_isstring(L, 2)) {
-        lua_pushstring(L, "permAlias: need a parent alias/group to add this alias to");
-        lua_error(L);
-        return 1;
-    } else {
-        luaParent = lua_tostring(L, 2);
+        lua_pushfstring(L, "permAlias: bad argument #2 type (alias parent as string expected, got %s!)",
+                        luaL_typename(L, 2));
+        return lua_error(L);
     }
+    QString parent = QString::fromUtf8(lua_tostring(L, 2));
 
-    string luaRegex;
     if (!lua_isstring(L, 3)) {
-        lua_pushstring(L, "permAlias: need the pattern for the alias");
-        lua_error(L);
-        return 1;
-    } else {
-        luaRegex = lua_tostring(L, 3);
+        lua_pushfstring(L, "permAlias: bad argument #3 type (regexp-type pattern as string expected, got %s!)",
```

Comment:
Do we still have the disabled comboBox locked to "regexp" on the GUI?  I though this was slightly more informative than just "regexp" though.

Reply:
Have changed it to `regexp pattern` - leaving the pattern word there because it might not be obvious that it does not _have_ to be a literal string...